pkg/channel/distributed/controller/kafkachannel/reconciler.go

```diff
@@ -206,17 +214,37 @@ func (r *Reconciler) reconcile(ctx context.Context, channel *kafkav1beta1.KafkaC
 	// ConfigMap.  Therefore, we will instead check the Kafka Secret associated with the
 	// KafkaChannel here.
 	//
-	if len(r.kafkaSecret) > 0 {
+	if len(r.config.Kafka.AuthSecretName) > 0 {
 		channel.Status.MarkConfigTrue()
 	} else {
 		channel.Status.MarkConfigFailed(event.KafkaSecretReconciled.String(), "No Kafka Secret For KafkaChannel")
 		return fmt.Errorf(constants.ReconciliationFailedError)
 	}
 
+	// Reconcile the Receiver Deployment/Service
+	secret, err := r.kubeClientset.CoreV1().Secrets(r.config.Kafka.AuthSecretNamespace).Get(ctx, r.config.Kafka.AuthSecretName, metav1.GetOptions{})
+	if err != nil && apierrs.IsNotFound(err) {
+		// The Receiver reconciler needs the namespace and name for various purposes, so construct a Secret with the required values
+		err = r.reconcileReceiver(ctx, channel, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: r.config.Kafka.AuthSecretName, Namespace: r.config.Kafka.AuthSecretNamespace}}, false)
+		if err != nil {
+			logging.FromContext(ctx).Error("Reconcile Receiver With Empty Secret Failed", zap.Error(err))
+		}
+
+		// Not having an auth secret for the Receiver is a problem
+		channel.Status.MarkConfigFailed(event.KafkaSecretReconciled.String(), "No Kafka Secret For KafkaChannel")
+		return fmt.Errorf(constants.ReconciliationFailedError)
+	} else if err != nil {
+		logging.FromContext(ctx).Error("Error reading receiver secret", zap.Error(err))
+		return fmt.Errorf(constants.ReconciliationFailedError)
+	}
+
+	receiverError := r.reconcileReceiver(ctx, channel, secret, true)
```

Comment:
`r.reconcileReceiver` is called 13 lines above with `exists=false` and I see this call is with `exists=true`. But do we need to call it again here?

Reply:
It's not really being called again here; if the "exists=false" path is taken, an early error-return from the function is guaranteed.
The purpose of the earlier call is to allow the reconciler to remove the receiver if the secret is missing - This is the original behavior of the multi-secret reconciler, though, and it actually may not even be important to delete the single receiver deployment/service if the secret is missing.  That would simplify this code quite a bit but would leave open the possibility of running receiver deployments that don't have a secret (which would lead to errors if that receiver needs to reconfigure itself).